modules/nw-metallb-configure-address-pool-vlan.adoc

```diff
@@ -0,0 +1,63 @@
+:_mod-docs-content-type: PROCEDURE
+[id="nw-metallb-configure-address-pool-vlan_{context}"]
+= Configure MetalLB address pool for VLAN
+
+As a cluster administrator, you can add address pools to your cluster to control the IP addresses on a created VLAN that MetalLB can assign to load-balancer services
+
+.Prerequisites
+
+* Install the OpenShift CLI (oc).
+
+* Separate VLAN configured.
+
+* Log in as a user with `cluster-admin` privileges.
+
+.Procedure
+
+. Create a file, such as `ipaddresspool-vlan.yaml`, with content such as the following example:
++
+[source,yaml]
+----
```

Comment:
@kquinn1204 The IP Address pool does not look complete in terms of vlan. Is it intended?

Note about global forwarding looks good.

Reply:
@asood-rh I was not sure what it should look like in terms of VLAN maybe a guess on my part. I thought it was enough to specify that the "IP range must match the subnet assigned to the VLAN on your network". What am I  missing.

Reply:
@kquinn1204 If we look at IP Address pool this how it looks complete YAML looks like
<pre>
apiVersion: metallb.io/v1beta1
kind: IPAddressPool
metadata:
  labels:
    zone: east
  name: ipaddresspool-l2
  namespace: metallb-system
spec:
  addresses:
  - 192.168.111.65-192.168.111.69
  - 192.168.111.70-192.168.111.74
  autoAssign: true
  avoidBuggyIPs: true
  serviceAllocation:
    namespaceSelectors:
    - matchLabels:
        environ: Test
    - matchExpressions:
      - key: region
        operator: In
        values:
        - NA
    namespaces:
    - test1
    - test2
    priority: 10
    serviceSelectors:
    - matchLabels:
        environ: Test
    - matchExpressions:
      - key: environ
        operator: In
        values:
        - Test
</pre>
If you mean to say the IP addresses belonging to different VLANs  then we would need to talk about connectivity. The requirement of L2 addresses is they need to be in same subnet as the nodes 's IP addresses.

Reply:
so I need to define two IP ranges, one for primary interface and one for configured VLAN @asood-rh

Reply:
@kquinn1204 You can define one range only as this is the range from where addresses are assigned to service. It is only this range if it is in VLAN should be reachable for the service to functional. IP address pool is just a pool of IP addresses from which addresses are assigned to a service, it is not aware of network topology.